jellyfin-platform-android/src/main/kotlin/org/jellyfin/apiclient/interaction/AndroidDevice.kt

```diff
@@ -9,13 +9,14 @@ import org.jellyfin.apiclient.model.DeviceInfo
 /**
  * Helper function used in [androidDevice] to normalize device names:
  *
- * - Removes special characters from the name.
+ * - Removes non US-ASCII printable characters from the name.
  * - Trims the whitespace at the start and end of the name.
+ * - Remove duplicate whitespace from the name.
  *
  * Returns a copy of the device with the normalized name.
  */
 fun DeviceInfo.normalize() = copy(
-	name = name.replace("[^\\w\\s]".toRegex(), "").trim()
+	name = name.replace("[^\\x20-\\x7E]".toRegex(), "").trim().replace("\\s{2,}".toRegex(), " ")
```

Comment:
Looks good, but maybe it'd make sense to split the lines and add some documentation to them, like so:
```suggestion
	name = name
	        .replace("[^\\x20-\\x7E]".toRegex(), "") // Remove non-ASCII-printable characters (only allow SPACE to ~)
	        .trim() // Remove surrounding whitespace
	        .replace("\\s{2,}".toRegex(), " ") // Replace repetitive whitespace
```

Reply:
Should I remove part of the comment or leave both comments? (comment block + comment in code)

https://github.com/jellyfin/jellyfin-apiclient-java/blob/bf9179cbf0a45461acb558ef0a9069c610e4bfb0/jellyfin-platform-android/src/main/kotlin/org/jellyfin/apiclient/interaction/AndroidDevice.kt#L12-L14

Reply:
Oh nevermind, should've read more carefully. It's fine then.